### PR TITLE
Ut3590/pythonnet dll linux

### PR DESCRIPTION
**Pull request by baptisteguidoux**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/pythonnet/pythonnet?pullRequest=1203) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/pythonnet/pythonnet?pullRequest=1203) before we can accept your contribution.<br/><br/>:x: baptisteguidoux [sign now](https://cla.dotnetfoundation.org/pythonnet/pythonnet?pullRequest=1203)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/pythonnet/pythonnet?pullRequest=1203) it.</sub>

File: src/runtime/pythonengine.cs [hunk 1]
```diff
@@ -146,6 +146,24 @@ public static int RunSimpleString(string code)
...
+        {
+            if (!libraryLoaded)
+            {
+#if MONO_OSX
+                var os = Python.Runtime.Platform.OperatingSystemType.Darwin;
+#elif MONO_LINUX
+                var os = Python.Runtime.Platform.OperatingSystemType.Linux;
+#else
+                var os = Python.Runtime.Platform.OperatingSystemType.Windows;
+#endif
```
Comment (filmor) [inline src/runtime/pythonengine.cs:160]: We are trying to get rid of compile-time checks for the OS. Please use equivalent runtime information, there should already be code in place on master to do this.

File: src/runtime/pythonengine.cs [hunk 2]
```diff
@@ -166,7 +193,7 @@ public static void Initialize(bool setSysArgv = true, bool initSigs = false)
         /// interpreter lock (GIL) to call this method.
         /// initSigs can be set to 1 to do default python signal configuration. This will override the way signals are handled by the application.
         /// </remarks>
-        public static void Initialize(IEnumerable<string> args, bool setSysArgv = true, bool initSigs = false)
+        static internal void Initialize2(IEnumerable<string> args, bool setSysArgv = true, bool initSigs = false)
```
Comment (filmor) [inline src/runtime/pythonengine.cs:196]: Please give this one a more telling name.

File: src/runtime/resources/clr.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 Code in this module gets loaded into the main clr module.
 """
 
-__version__ = "2.5.0"
+__version__ = "2.5.1"
```
Comment (filmor) [inline src/runtime/resources/clr.py:5]: Please exclude the 2.5.1 commits. The actual changes from that branch should already be in master (if not, make them a separate PR, the update to the Changelog makes sense).

File: src/runtime/runtime.cs [hunk 4]
```diff
@@ -95,12 +95,18 @@ public class Runtime
...
+#if MONO_LINUX || MONO_MAC
+        internal const string dllDirectory = "Library/PythonInstall/lib/";
+        internal const string pythonlib = "python3.7m";
+// #elif MONO_MAC
+//         internal const string dllDirectory = "Library/PythonInstall/lib/";
+//         internal const string pythonlib = "python3.7m";
+#else //windows
+        internal const string dllDirectory = "Library/PythonInstall/";
+        internal const string pythonlib = "python37";
 #endif
```
Comment (filmor) [inline src/runtime/runtime.cs:109]: This is all very Unity specific. Maybe just allow `dllDirectory` to be set at build-time or something like that?

**Comment by baptisteguidoux** [comment]
Sorry, wrong repo, closing this PR.
